cluster/engine.go

```diff
@@ -310,9 +310,11 @@ func (e *Engine) HealthIndicator() int64 {
 }
 
 // setState sets engine state
-func (e *Engine) setState(state engineState) {
-	e.Lock()
-	defer e.Unlock()
+func (e *Engine) setState(state engineState, lock bool) {
+	if lock {
```

Comment:
I am wondering what is the use of this parameter `lock` passed in func `setState`?
I think
```
e.Lock()
defer e.Unlock() 
```
is OK for me.
@xianlubird